### DISCUSSION TITLE
Next Headers Config Disallows Capture Groups

### DISCUSSION THREAD

**Discussion by caweinshenker** [vercel/next.js#52294] [Help]
### Summary

I'm trying to write a header to match all font files in `/_next/static/media`. I've written `source` as `/_next/static/media/:file((\\.woff|\\.woff2|\\.ttf))`, but `next build` errors with `Capturing groups are not allowed`

```
Error parsing `/_next/static/media/:file((\.woff|\.woff2|\.ttf))` https://nextjs.org/docs/messages/invalid-route-source
Reason: Capturing groups are not allowed at 26

  /_next/static/media/:file((\.woff|\.woff2|\.ttf))
```

Is it possible to do a "one-of" match using the `next.config` `headers` `source` option?

### Additional information

```js
headers: async () => [
    {
      headers: [
        {
          key: 'Access-Control-Allow-Origin',
          value: '*',
        },
        {
          key: 'Access-Control-Allow-Methods',
          value: 'GET, OPTIONS',
        },
      ],
      source: '/_next/static/media/:file((\\.woff|\\.woff2|\\.ttf))',
    },
  ],
```
```


### Example

_No response_

**Comment by rasif-sahl** [comment]
The error message you received indicates that capturing groups are not allowed in the source pattern used in the next.config.js file for the headers option. To achieve a "one-of" match for font files in /static/media, you can modify your configuration as follows:

```
// next.config.js

module.exports = {
  async headers() {
    return [
      {
        source: '/_next/static/media/:file(.*\\.(woff|woff2|ttf))',
        headers: [
          {
            key: 'Access-Control-Allow-Origin',
            value: '*',
          },
          {
            key: 'Access-Control-Allow-Methods',
            value: 'GET, OPTIONS',
          },
        ],
      },
    ];
  },
};
```

In the modified configuration, the regular expression .*\\.(woff|woff2|ttf) is used to match any file that ends with .woff, .woff2, or .ttf within the /static/media directory. The backslashes (\\) are used to escape the dot (.) in the regular expression.

By using this updated configuration, you should be able to set the headers for font files in /static/media as expected.

Please note that this configuration is specific to setting headers and does not affect the routing or file serving behavior in Next.js. Ensure that your server is correctly configured to handle the headers according to your requirements.

**Reply by caweinshenker** [reply]
This doesn't  work either. The issue seems to be that the "one-of" syntax has precisely the same form as a capture group

```
Error parsing `/_next/static/media/:file(.*\.(woff|woff2|ttf))` https://nextjs.org/docs/messages/invalid-route-source
Reason: Capturing groups are not allowed at 30

  /_next/static/media/:file(.*\.(woff|woff2|ttf))
                                ^
 ```
